### PR TITLE
News8

### PR DESCRIPTION
**Pull request by Castelli87**

### CODE REVIEW

File: src/Components/ArticleMainCard.jsx [hunk 1]
```diff
@@ -36,6 +41,37 @@ function ArticleMainCard() {
     })
   };
 
+////////////////UPLIFTING FROM THE CHILD COMPONENT/////////////////////////////////
```
Comment (aloush-dev) [inline src/Components/ArticleMainCard.jsx:47]: was there a reason you pulled it out of your comments list component as this component is getting large now

File: src/Components/ArticleMainCard.jsx [hunk 2]
```diff
@@ -36,6 +41,37 @@ function ArticleMainCard() {
...
+    if (newComment.length === 0){
+      return e.preventDefault()
+    }
+
+    postComment(article_id,newComment,user.username).then((res)=>{
+      setPostError(false)
+      setNewComment('')
+      setComments((currentComments)=>{ 
+        return [res,...currentComments ]
+    })
```
Comment (aloush-dev) [inline src/Components/ArticleMainCard.jsx:91]: this being in a .then block after the api call means it's not optimistically rendered

File: src/Components/ArticleMainCard.jsx [hunk 3]
```diff
@@ -36,6 +41,37 @@ function ArticleMainCard() {
...
+ 
+    }).catch(()=>{
+      setComments((currentComments)=>{
+        setPostError(true)
+        return currentComments
+      })
```
Comment (aloush-dev) [inline src/Components/ArticleMainCard.jsx:97]: you need to remove the comment which from the array after you've changed it to be optimistically rendered

**Review by aloush-dev — CHANGES_REQUESTED** [review]
Look into making sure somebody can only make a comment once, I was able to spam the post button and it posted multiple times

File: src/Components/ArticleMainCard.jsx (hunk 1 above)
Comment (Castelli87) [reply src/Components/ArticleMainCard.jsx:47]: Before the api call and the state of commentsList were  in the child component  but now  i thought that i need on the parent child to spread into the list, but I'm sure how fix this stuff so i dropped a msg on slack let me know if is possible have a zoom call with you

File: src/Components/ArticleMainCard.jsx (hunk 3 above)
Comment (Castelli87) [reply src/Components/ArticleMainCard.jsx:97]: Thanks for the feedback but i need some guidance please

**Pull request merged**